### PR TITLE
STUDENT - sending data imported from the student's file to the frontend

### PR DESCRIPTION
**Pull request by mae1579**
STUDENT - sending data imported from the student's file to the frontend

Signed-off-by: Mateusz Umięcki <mateuszumiecki@gmail.com>

### CODE REVIEW

File: routers/get.student.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { pool } from '../utils/db';
+import { FieldPacket } from 'mysql2';
+import { SingleStudentImport } from '../types';
+import { ValidationError } from '../utils/handleErrors';
+
+export const getStudentRouter = Router();
+type SingleStudentEntity = [SingleStudentImport[], FieldPacket[]];
+getStudentRouter.get('/:userId', async (req, res) => {
+  if (!req.params.userId) {
+    throw new ValidationError('Nie otrzymano user id !');
```
Comment (patrycja-dev41171) [inline routers/get.student.ts:11]: There should be no 'space' in front of the exclamation mark.

File: routers/get.student.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  const [result] = (await pool.execute(
+    'SELECT `student`.`courseCompletion`, `student`.`courseEngagement`, `student`.`projectDegree`, `student`.`teamProjectDegree`, `student`.`bonusProjectUrls`,`user`.`email` FROM `user` INNER JOIN `student` on `user`.`id` = `student`.`user_id` WHERE `user`.`id` = :id',
+    {
+      id: req.params.userId,
+    }
+  )) as SingleStudentEntity;
+  const data = result.length === 0 ? null : result[0];
+  if (data === null || data === undefined) {
+    throw new ValidationError('Nie udało się odszukać danych o zaimportowanym studencie  !');
```
Comment (patrycja-dev41171) [inline routers/get.student.ts:22]: !

File: types/student/student-entity.ts [hunk 3]
```diff
@@ -29,6 +29,15 @@ export interface StudentImport {
...
 
+export interface SingleStudentImport {
+  email: string;
+  courseCompletion: number;
+  courseEngagement: number;
+  projectDegree: number;
+  teamProjectDegree: number;
+  bonusProjectUrls: string;
+}
+
```
Comment (patrycja-dev41171) [inline types/student/student-entity.ts:40]: I think that the student id could also be very useful on FE, when redirecting from the list of students to the profile of a specific student using the id in the URL.

**Review by patrycja-dev41171 — COMMENTED** [review]
This code needs some minor corrections, but still well done!

File: routers/get.student.ts (hunk 2 above)
Comment (LukaszAmbrozewski) [inline routers/get.student.ts:22]: You can remove unnecessary separators.

File: types/student/student-entity.ts (hunk 3 above)
Comment (LukaszAmbrozewski) [inline types/student/student-entity.ts:40]: Good job!

**Pull request merged**